source/geometry_model/initial_topography_model/prm_polygon.cc

```diff
@@ -0,0 +1,145 @@
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/geometry_model/initial_topography_model/prm_polygon.h>
+#include <aspect/utilities.h>
+#include <boost/lexical_cast.hpp>
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    double
+    PrmPolygon<dim>::
+    value (const Point<dim-1> &p) const
+    {
+      double value = 0;
+      for (unsigned int i = 0; i < point_lists.size(); i++)
+        {
+          Point<2> p1(p[0],p[1]);
+
+          if (aspect::Utilities::in_2d_polygon<dim>(p1, point_lists[i]))
+            {
+              value = topography_values[i];
+            }
+        }
+      return value;
+    }
+
+
+
+    template <int dim>
+    void
+    PrmPolygon<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography");
+        {
+          prm.enter_subsection("Prm polygon");
+          {
+            prm.declare_entry("Topography parameters",
+                              "",
+                              Patterns::Anything(),
+                              "Set the topography height, end with a |, and set the areas described "
+                              "by the points, separated by commas and coordinates separated by a ':'. "
+                              "Seperate each topography feature by a semicolon. For example for two triangular areas of "
+                              "100 and -100 meters high set: '100|0:0,5:5,0:10;-100|10:10,10:15,20:15'. "
+                              "Units are dependent on the geometry model.");
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    PrmPolygon<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography");
+        {
+          prm.enter_subsection("Prm polygon");
+          {
+            /**
+             * we need to fill the point lists and topography values. They
+             * are stored in the Topography subsection in the Topography parameter.
+             */
+
+            std::string temptopo = prm.get("Topography parameters");
+            std::vector<std::string> temp_topographies = Utilities::split_string_list(temptopo,';');
+            unsigned int temp_topographies_size = temp_topographies.size();
+
+            topography_values.resize(temp_topographies_size,0);
+            point_lists.resize(temp_topographies_size);
+            for (unsigned int i_topo = 0; i_topo < temp_topographies_size; i_topo++)
+              {
+                std::vector<std::string> temp_topography = Utilities::split_string_list(temp_topographies[i_topo],'|');
+                Assert(temp_topography.size() == 2,ExcMessage ("The given line '" + temp_topographies[i_topo] + "' is not correct. "
```

Comment:
Can you break the ExcMessage onto the next line to shorten the length of the lines here?